bindings/matrix-sdk-crypto-nodejs/src/attachment.rs

```diff
@@ -0,0 +1,98 @@
+use std::{
+    io::{Cursor, Read},
+    ops::Deref,
+};
+
+use napi::bindgen_prelude::Uint8Array;
+use napi_derive::*;
+
+use crate::into_err;
+
+/// A type to encrypt and to decrypt anything that can fit in an
+/// `Uint8Array`, usually big buffer.
+#[napi]
+pub struct Attachment;
+
+#[napi]
+impl Attachment {
+    /// Encrypt the content of the `Uint8Array`.
+    ///
+    /// It produces an `EncryptedAttachment`, we can be used to
+    /// retrieve the media encryption information, or the encrypted
+    /// data.
+    #[napi]
+    pub fn encrypt(array: Uint8Array) -> napi::Result<EncryptedAttachment> {
+        let buffer: &[u8] = array.deref();
+
+        let mut cursor = Cursor::new(buffer);
+        let mut encryptor = matrix_sdk_crypto::AttachmentEncryptor::new(&mut cursor);
+
+        let mut encrypted_data = Vec::new();
+        encryptor.read_to_end(&mut encrypted_data).map_err(into_err)?;
```

Comment:
It's a bit of a shame that everywhere we end up using the `AttachmentEncryptor` we end up just using `read_to_end()`.

I guess the problem on the native side is that the Ruma request doesn't support that quite well, `reqwest` should be able to handle this: https://docs.rs/reqwest/latest/reqwest/blocking/multipart/struct.Part.html#method.reader.

Anyways, sorry for the digression here, there's nothing to do on the bindings side.